pages/library/_viewId.vue

```diff
@@ -157,6 +198,127 @@ export default Vue.extend({
     }
   },
   methods: {
+    async filterMedia() {
+      try {
+        const collectionInfo = await this.$api.items.getItems({
+          uId: this.$auth.user.Id,
+          userId: this.$auth.user.Id,
+          ids: this.$route.params.viewId
+        });
+
+        if (
+          collectionInfo.data.Items &&
+          (collectionInfo.data.Items[0].Type === 'CollectionFolder' ||
+            collectionInfo.data.Items[0].Type === 'Folder')
+        ) {
+          const options: any = {
+            uId: this.$auth.user.Id,
+            userId: this.$auth.user.Id,
+            parentId: this.$route.params.viewId,
+            includeItemTypes: '',
+            recursive: true,
+            sortBy: this.orderMethod,
+            sortOrder: this.sortDirection,
+            fields: this.orderMethod
+          };
+
+          if (collectionInfo.data.Items[0].CollectionType === 'tvshows') {
+            options.includeItemTypes = 'Series';
+          } else if (collectionInfo.data.Items[0].CollectionType === 'movies') {
+            options.includeItemTypes = 'Movie';
+          } else if (collectionInfo.data.Items[0].CollectionType === 'books') {
+            options.includeItemTypes = 'Book';
+          }
+
+          const filterString = this.makeFilterString(
+            this.allFilters.filters.items,
+            ','
+          );
+          options.filters = filterString;
+
+          for (let i = 0; i < this.allFilters.features.items.length; i++) {
```

Comment:
Use a modern type of `for` to do this:
```suggestion
          for (const feature of this.allFilters.features.items) {
```

They're generally more readable.

Reply:
You can also probably use `.filter()` ([MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)) instead of a loop, to make it even nicer.